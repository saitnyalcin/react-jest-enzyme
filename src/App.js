import React from 'react';
import './App.css';
import Footer from './Footer';
import logo from './images/gigasecond-white.png';
import MyComponent from './MyComponent';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} alt="" />
        <div className="header">Jest & Enzyme testing</div>
      </header>

      <MyComponent />
      <Footer />
    </div>
  );
}

export default App;
