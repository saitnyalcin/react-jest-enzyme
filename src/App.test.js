import { shallow } from 'enzyme';
import React from 'react';
import App from './App';

describe('rendering App component and its nested elements', () => {
  let component;
  beforeEach(() => {
    component = shallow(<App />);
  });

  it('rendering the div header in App component', () => {
    expect(component.find('div.header').text()).toBe('Jest & Enzyme testing');
  });

  it('rendering the nested component inside the App component', () => {
    expect(
      component.find('MyComponent').exists() &&
        component.find('Footer').exists()
    ).toBe(true);
  });

  it('snapshot the App component', () => {
    expect(component).toMatchSnapshot();
  });
});
