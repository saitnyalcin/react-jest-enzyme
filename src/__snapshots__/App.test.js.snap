// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot the component snapshoting the App component 1`] = `ShallowWrapper {}`;

exports[`App component snapshoting the App component 1`] = `ShallowWrapper {}`;

exports[`rendering App component and its nested elements snapshot the App component 1`] = `ShallowWrapper {}`;

exports[`rendering App component and its nested elements snapshoting the App component 1`] = `ShallowWrapper {}`;

exports[`snapshot the component snapshoting the App component 1`] = `ShallowWrapper {}`;
